internal/metastore/kv/rootcoord/suffix_snapshot.go

```diff
@@ -543,3 +553,80 @@ func (ss *SuffixSnapshot) MultiSaveAndRemoveWithPrefix(saves map[string]string,
 	}
 	return err
 }
+
+func (ss *SuffixSnapshot) Close() {
+	close(ss.closeGC)
+}
+
+// startBackgroundGC the data will clean up if key ts!=0 and expired
+func (ss *SuffixSnapshot) startBackgroundGC() {
+	log.Debug("suffix snapshot GC goroutine start!")
+	ticker := time.NewTicker(15 * time.Minute)
+	defer ticker.Stop()
+
+	params := paramtable.Get()
+	retentionDuration := params.CommonCfg.RetentionDuration.GetAsDuration(time.Second)
+
+	for {
+		select {
+		case <-ss.closeGC:
+			log.Warn("quit suffix snapshot GC goroutine!")
+			return
+		case now := <-ticker.C:
+			err := ss.removeExpiredKvs(now, retentionDuration)
+			if err != nil {
+				log.Warn("remove expired data fail during GC", zap.Error(err))
+			}
+		}
+	}
+}
+
+func (ss *SuffixSnapshot) removeExpiredKvs(now time.Time, retentionDuration time.Duration) error {
+	removedKeyPrefix := make(map[string]struct{})
+
+	err := ss.TxnKV.WalkWithPagination(ss.snapshotPrefix, PaginationSize, func(k []byte, v []byte) error {
+		key := string(k)
+		value := string(v)
+		// skip if the key hasn't tombstone
+		if !ss.isTombstone(value) {
+			return nil
+		}
+
+		key = ss.hideRootPrefix(key)
+		ts, ok := ss.isTSKey(key)
+		// skip if the key doesn't contain ts
+		if !ok {
+			return nil
+		}
+
+		pts, _ := tsoutil.ParseTS(ts)
+		expireTime := pts.Add(retentionDuration)
+		// skip if it doesn't reach expire time
+		if now.Before(expireTime) {
+			return nil
+		}
+
+		// collect keys that parent node is snapshot node if the corresponding the latest ts is expired.
+		idx := strings.LastIndex(key, ss.separator)
```

Comment:
if we have following data:
A/100
A/105 Delete
A/120 
the data before 105 is expired. what prefix will we generate? Did we test on it?
The expect output should be delete data between A/100 - A/105

Reply:
Yes,  the test cases have covered these scenario